snap/squashfs/squashfs_test.go

```diff
@@ -962,3 +975,27 @@ func (s *SquashfsTestSuite) TestBuildWithCompressionUnhappy(c *C) {
 	})
 	c.Assert(err, ErrorMatches, "(?m)^mksquashfs call failed: ")
 }
+
+func (s *SquashfsTestSuite) TestBuildBelowMinimumSize(c *C) {
+	// this snap is empty. without truncating it to be larger, it should be smaller than
+	// the minimum snap size
+	sn := squashfs.New(filepath.Join(c.MkDir(), "truncate_me.snap"))
+	sn.Build(c.MkDir(), nil)
+
+	size, err := sn.Size()
+	c.Assert(err, IsNil)
+
+	c.Assert(size, Equals, squashfs.MinimumSnapSize)
+}
+
+func (s *SquashfsTestSuite) TestBuildAboveMinimumSize(c *C) {
+	// fill a snap with random data that will not compress well. it should be forced
+	// to be bigger than the minimum threshold
+	randomData := randutil.RandomString(int(squashfs.MinimumSnapSize * 2))
```

Comment:
My opinion is that the data should be logged on test failure if it needs to be random, otherwise we have no way of reproducing something that would make it fail.

Reply:
That makes sense, but that would be 32768 characters to log. Is that okay?

Reply:
You can format it as a [Comment](https://pkg.go.dev/gopkg.in/check.v1?utm_source=godoc#Commentf) and pass it as a fourth argument to the assertion, so it's only written if the assertion fails